### DISCUSSION TITLE
Execute py file not using `mpiexec -n num_process python main.py`

### DISCUSSION THREAD

**Discussion by Qiang-CU** [mpi4py/mpi4py#245] [Q&A] [answered]
Hello,

I am working on some codes using mpi4py and I have the following 2 questions. 

1. Can I run main.py in parallel **not** using a terminal? I usually use pycharm to program and I hope to run my file using the green triangle button on the pycharm interface rather than the terminal.

2. I want to create 10 processes and divide them into three groups, id = [(0,1,2,3,4,5), (6,7), (8,9)]. For each group, I want to make them finish one task collaboratively, i.e., processes id (0,1,2,3,4,5) work on task 1, id (6,7) work on task 2, id(8,9) work on task 3. And I hope that processes in each group can communicate with each other rather than process which does not belong to this group. e.g., id 0 can communicate with 1-5, not 6,7, nor 8,9. Is there any method in mpi4py that can help me?

Thank you very much.

Best Regards,

Qiang

**Comment by dalcinl** [comment, marked as answer]
1. Of course it can, but it is usually outside of mpi4py's control and/or responsibilities. You should look at PyCharm's documentation to figure out if you can configure/edit the command used to execute scripts, and whether you an do execute the Python interpreter via `mpiexec`. If that's not possible our you cannot figure it out, you should contact PyCharm's developers. It is not possible to run MPI codes in parallel without the intervention of a launcher that orchestrates the spawn and handshaking of these processes, the `mpiexec` command must be involved.
2. MPI provides various ways to split a parent communicator (usually `MPI.COMM_WORLD`) in subcommunicators. The easiest to use is probably the `MPI.Intacomm.Split()` method. Afterwards, these subcommunicators can be linked together (in pairs) using intercommunicators (class `MPI.Intercomm` in mpi4py) using `MPI.Intacomm.Create_intercomm()`. Read the MPI standard or MPI books to understand how all that stuff works.
